libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +83,258 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
   public buttonContainerElement: ElementRef<HTMLElement> | undefined;
 
   @ViewChild('defaultButton', { read: ElementRef })
-  public defaultButtonElement: ElementRef<HTMLElement> | undefined;
+  public defaultButtonElement: ElementRef<HTMLButtonElement> | undefined;
 
   @ContentChild(ButtonComponent, { read: ElementRef }) public userProvidedButton:
-    | ElementRef<HTMLElement>
+    | ElementRef<HTMLButtonElement>
     | undefined;
 
   @ViewChild(FloatingDirective)
-  private floatingDirective: FloatingDirective;
+  private floatingMenu: FloatingDirective;
 
-  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
+  @ContentChildren(ItemComponent, { read: ElementRef }) public kirbyItems: QueryList<
+    ElementRef<HTMLElement>
+  >;
+
+  @ContentChildren(ItemComponent) public kirbyItemComponents: QueryList<ItemComponent>;
 
+  public floatingMenuIsShown: boolean = false;
+  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
   private scrollListenerDisposeFn: EventListenerDisposeFn;
+  private focusedIndex = -1;
+
+  @HostListener('keydown', ['$event'])
+  _onKeydown(event: KeyboardEvent) {
+    if (this.kirbyItems.length === 0) {
+      console.warn('No items found within menu');
```

Comment:
Consider prefixing to improve guidance from log entry
```suggestion
      console.warn('[Kirby] No items found within menu');
```